src/SimulationRunner.cc

```diff
@@ -559,6 +560,8 @@ void SimulationRunner::ProcessSystemQueue()
 
     this->postUpdateThreads.push_back(std::thread([&, id]()
     {
+      auto parentEntity = system.parent;
+      auto this_thread_system = system.system;
```

Comment:
I think (but I'm not 100% sure) another option is to change the ranged for loop starting on line 557 to create the `system` variable as a copy instead of a reference by deleting `&` from that line, but this approach works. I'll be ready to approve once the linter is happy

Reply:
Yet another way would be to stop all systems upon removal, just like we do when we spawn systems. This would reduce a significant amount of code. I'll hack it out this week.